src/srf/boolean.cpp

```diff
@@ -20,20 +20,23 @@ void SShell::MakeFromIntersectionOf(SShell *a, SShell *b) {
     MakeFromBoolean(a, b, SSurface::CombineAs::INTERSECTION);
 }
 
-// We will be inserting existing verticies into curves to split them
-// todo: this is only using the ends of exact curves, and it is only
-// using them to split existing curves, not new intersections.
-// It resolves some issues but we could do better. We will need to
-// reorder things so the surface intersection curves exist prior to
-// splitting any curves at all in order to have their verticies too.
-// Better still would be to get curve/surface intersection to work
```

Comment:
Leave some more of the explanation in. Otherwise it will get buried in the history and it is important.